core/src/main/scala/com/typesafe/tools/mima/lib/analyze/method/MethodChecker.scala

```diff
@@ -47,14 +47,15 @@ private[analyze] object MethodChecker {
         hasMatchingSignature(oldmeth.signature, newmeth.signature, newmeth.bytecodeName)
   }
 
-  private[analyze] def hasMatchingSignature(oldsig: String, newsig: String, bytecodeName: String): Boolean = {
-    oldsig == newsig || {
-      // Special case for https://github.com/scala/scala/pull/7975:
-      // uses .tail to drop the leading '(' in the signature
-      bytecodeName == MemberInfo.ConstructorName && (newsig.isEmpty || oldsig.endsWith(newsig.tail))
-    }
+  private def hasMatchingSignature(oldsig: String, newsig: String, bytecodeName: String): Boolean = {
+    oldsig == newsig || // Special case for scala#7975
+      bytecodeName == MemberInfo.ConstructorName && hasMatchingCtorSig(oldsig, newsig)
```

Comment:
(I'd be fine with moving `bytecodeName == MemberInfo.ConstructorName` to the helper function as well)

Reply:
Purposely kept it out, to avoid needing to pass it in the test.  It's part of the method name.